### PR TITLE
UCM: Make ucm_reloc_get_orig() a static function.

### PR DESCRIPTION
**Pull request by alinask**

### CODE REVIEW

**Comment by alinask** [comment]
@bureddy @yosefe Please review.
Required to fix the problem described in https://github.com/openucx/ucx/pull/3427

File: src/ucm/util/reloc.h [hunk 1]
```diff
@@ -39,12 +41,32 @@ ucs_status_t ucm_reloc_modify(ucm_reloc_patch_t* patch);
...
+ * This function is set to inline to make sure that the symbol search done here
+ * is done from within the shared object it was invoked from.
+ *
  * @param [in]  symbol       Symbol name,
  * @param [in]  replacement  Symbol replacement, which should be ignored.
  *
  * @return Original function pointer for 'symbol'.
  */
-void* ucm_reloc_get_orig(const char *symbol, void *replacement);
+static inline void* ucm_reloc_get_orig(const char *symbol, void *replacement)
```
Comment (evgeny-leksikov) [inline src/ucm/util/reloc.h]: UCS_F_ALWAYS_INLINE?

**Comment by alinask** [comment]
@evgeny-leksikov done, thanks!

File: src/ucm/util/reloc.h [hunk 2]
```diff
@@ -10,6 +10,8 @@
 
 #include <ucs/datastruct/list.h>
 #include <ucs/type/status.h>
+#include <ucm/util/log.h>
+#include <dlfcn.h>
```
Comment (evgeny-leksikov) [inline src/ucm/util/reloc.h]: These includes can be removed from .c

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9613/ for details (Mellanox internal link).

File: src/ucm/util/reloc.h [hunk 3]
```diff
@@ -39,12 +41,33 @@ ucs_status_t ucm_reloc_modify(ucm_reloc_patch_t* patch);
 /**
  * Get the original implementation of 'symbol', which is not equal to 'replacement'.
  *
+ * This function is set to static to make sure that the symbol search done here
```
Comment (yosefe) [inline src/ucm/util/reloc.h]: This function is static to make sure the symbol search is done from the context of the shared object which defines the replacement function. If the replacement function is defined in a loadbale module, the symbols it imports from other libraries may not be available in global scope.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9614/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9615/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9617/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9652/ for details (Mellanox internal link).

**Comment by alinask** [comment]
@yosefe Please review.

**Comment by bureddy** [comment]
@yosefe @alinask I think this should also goto 1.6.x. WDYT?

**Comment by yosefe** [comment]
yes, v1.6.x

**Comment by alinask** [comment]
Sure, I'll commit to v1.6.x as well.

**Comment by alinask** [comment]
@yosefe @bureddy https://github.com/openucx/ucx/pull/3447

**Pull request merged**